test/run.js

```diff
@@ -0,0 +1,304 @@
+        } else {
+            setImmediate(cb, null);
+        }
+    };
+}
+
+function setupOutdatedModules(obj, testsExpectToPass) {
+    return function () {
+        execBackup = childProcess.exec;
+        childProcess.exec = execMock(obj, testsExpectToPass);
+    };
+}
+
+describe("run()", function () {
+    describe(".emit('init')", function () {
+        before(setupOutdatedModules(outdatedModules));
+
+        after(tearDown);
+
+        it("should throw an error, if no options set", function () {
+            expect(run).to.throw(Error);
+        });
+
+        it("should throw an error, if cwd is missing", function () {
+            expect(
+                function () {
+                    run({ cwd: 1 });
+                }
+            ).to.throw(Error);
+        });
+
+        it("should be emitted, if cwd is set", function (done) {
+            reporter = function (emitter) {
+                emitter.on("init", function (options) {
+                    expect(options).to.eql({ cwd: "/Users/peerigon/Workspace/peerigon/updtr" });
+                });
+            };
+
+            run({ cwd: process.cwd(), reporter: reporter }, done);
+        });
+    });
+
+    describe(".emit('noop')", function () {
+        before(setupOutdatedModules(noOutdatedModules));
+        afterEach(tearDown);
+
+        it("should be emitted if no outdated modules were found", function (done) {
+            reporter = function (emitter) {
+                emitter.on("noop", function (options) {
+                    expect(options).to.be.undefined; // eslint-disable-line
+                });
+            };
+
+            run({ cwd: process.cwd(), reporter: reporter }, done);
+        });
+    });
+
+    describe(".emit('outdated')", function () {
+        describe("without git modules", function () {
+            before(setupOutdatedModules(outdatedModules));
+            afterEach(tearDown);
+
+            it("should be emitted if outdated modules were found", function (done) {
+                reporter = function (emitter) {
+                    emitter.on("outdated", function (options) {
+                        expect(options).to.eql(expectedOptions);
+                    });
+                };
+
+                run({ cwd: process.cwd(), reporter: reporter }, done);
+            });
+        });
+
+        describe("whit git modules", function () {
+            before(setupOutdatedModules(outdatedModulesWithGitDependencies));
+            afterEach(tearDown);
+
+            it("should be emitted without git modules", function (done) {
+                reporter = function (emitter) {
+                    emitter.on("outdated", function (options) {
+                        expect(options).to.eql(expectedOptions);
+                    });
+                };
+
+                run({ cwd: process.cwd(), reporter: reporter }, done);
+            });
+        });
+    });
+
+    describe(".emit('updating')", function () {
```

Comment:
You don't describe the `.emit()` method (for instance, you have also tests when `emit()` should *not* be executed). You're describing events, that why you should imho have a `describe("events", ...)` block with events name like `describe("updating", ...)`